lib/charms/prometheus_pushgateway_k8s/v0/pushgateway.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details
+
+r"""Interface library for the Prometheus Pushgateway.
+
+This library wraps a relation endpoint using the `pushgwateway` interface
+and exposes an API for forwarding metrics to Prometheus.
+
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.prometheus_pushgateway_k8s.v0.pushgateway
+```
+
+In the `metadata.yaml` of the charm, add the following:
+
+```yaml
+requires:
+  your-relation-name:
+    interface: pushgateway
+```
+
+In the source of your charm, first import the interface:
+
+```
+from charms.prometheus_pushgateway_k8s.v0.pushgateway import PrometheusPushgatewayRequirer
+````
+
+Instantiate the object in your charm's `__init__`, like so:
+
+```
+from charms.prometheus_pushgateway_k8s.v0.pushgateway import PrometheusPushgatewayRequirer
+from ops.charm import CharmBase
+
+class MyCharm(CharmBase):
+    def __init__(...):
+         ...
+        self.pushgateway_requirer = PrometheusPushgatewayRequirer(self, "your-relation-name")
+```
+
+Then use it at any moment to send a metric (validating that the requirer is ready), passing
+its name and value:
+
+```
+    if self.pushgateway_requirer.is_ready():
+        self.pushgateway_requirer.send_metric("test_metric", 3.141592)
```

Comment:
I wonder if we will be better off swapping `ignore_error` default to True (best-effort and silently fail). It seems that in the juju model it would make more sense to:
- Block the charm, informing about a missing relation
- Let the user have a nice API without checks

```suggestion
    self.pushgateway_requirer.send_metric("test_metric", 3.141592)
```

Reply:
I prefer to never silently fail by default. "Errors should never pass silently, unless explicitly silenced."

Regarding blocking the charm, this is more difficult now than before (because previously we relied on events, and now the relation data is dynamically checked). So for example if you want to send a metric and ask for readiness, and as it's not ready you block the charm, when you unblock it? Check for readiness on every event? Etc. Let's discuss this approach after this PR landed and dust settled a little.